src/frontend/Player/videojs/qualitySelectorPlugin/index.ts

```diff
@@ -0,0 +1,79 @@
+import videojs from 'video.js';
+
+import { QualitySelectorMenuButton } from './components/QualitySelectorMenuButton';
+import { QualitySelectorMenuItem } from './components/QualitySelectorMenuItem';
+import { Events, QualitySelectorMenuItemOptions } from './types';
+import './middleware';
+
+const Plugin = videojs.getPlugin('plugin');
+
+export class QualititySelector extends Plugin {
+  constructor(player: videojs.Player, options?: any) {
+    super(player, options);
+
+    videojs.registerComponent(
+      'QualitySelectorMenuButton',
+      QualitySelectorMenuButton,
+    );
+    videojs.registerComponent(
+      'QualitySelectorMenuItem',
+      QualitySelectorMenuItem,
+    );
+    if (options && options.default) {
```

Comment:
This beautiful optional chaining operator is for you `if(options?.default) { ...`